changedetectionio/tests/test_xpath_selector_unit.py

```diff
@@ -201,3 +201,56 @@ def test_trips(html_content, xpath, answer):
     html_content = html_tools.xpath_filter(xpath, html_content, append_pretty_line_formatting=True)
     assert type(html_content) == str
     assert answer in html_content
+
+DOM_violation_two_html_root_element = """<!DOCTYPE html>
+<html>
+  <body>
+    <h1>Hello world</h1>
+    <p>First paragraph.</p>
+  </body>
+</html>
+<html>
+  <body>
+    <h1>Hello world</h1>
+    <p>Browsers parse this part by fixing it but lxml doesn't and returns two root element node</p>
+    <p>Therefore, if the path is /html/body/p[1], lxml(libxml2) returns two element nodes not one.</p>
+  </body>
+</html>"""
+@pytest.mark.parametrize("html_content", [DOM_violation_two_html_root_element])
+@pytest.mark.parametrize("xpath, answer", [
+    ("/html/body/p[1]", "First paragraph."),
+    ("/html/body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
```

Comment:
This is the critical point. why do I choose one element in the browser inspect window, but lxml returns two? Because there are two html tag elements and two body tag elements.